org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/core/compiler/CharOperation.java

```diff
@@ -3734,59 +3734,15 @@ public static final void replace(char[] array, char[] toBeReplaced, char replace
  * @throws NullPointerException if the given array is null
  */
 public static final char[] replace(
-	char[] array,
-	char[] toBeReplaced,
-	char[] replacementChars) {
-
-	int max = array.length;
-	int replacedLength = toBeReplaced.length;
-	int replacementLength = replacementChars.length;
-
-	int[] starts = new int[5];
-	int occurrenceCount = 0;
-
-	if (!equals(toBeReplaced, replacementChars)) {
-
-		next : for (int i = 0; i < max;) {
-			int index = indexOf(toBeReplaced, array, true, i);
-			if (index == -1) {
-				i++;
```

Comment:
The actual bug is here: If index == -1, there is not hit. No need to scan again from the next offset.

What I like about this impl here: it's garbage-free whereas the proposed solution needs to encode the char arrays as ISO or UTF-8.

Do you have time to measure the pressure on the GC if your version of replace is used in tight loops?

Reply:
agree, that sounds like a better fix. One should compare the measurement for both solutions.

Reply:
If you guys are happier with a smaller change, then I'm fine with it.  I'll just abort the loop at that point and be done with it.

Reply:
using "break" instead of "continue" and removing the "next" label would be prettier, but i am OK with this fix